pkg/ingester/tailer.go

```diff
@@ -219,6 +222,10 @@ func (t *tailer) dropStream(stream logproto.Stream) {
 		t.blockedAt = &blockedAt
 	}
 
+	if len(t.droppedStreams) > maxDroppedStreams {
+		t.droppedStreams = nil
```

Comment:
Should we log this? Add a metric?

Reply:
In fact the idea of having new metric came up while discussing it with @slim-bean. But then we felt like new metric won't have much value particularly after making `droppedStreams` slice bounded.

But logging make sense here. I will add it.